pkg/compactor/compactor.go

```diff
@@ -1148,3 +1224,20 @@ func (c *Compactor) isPermissionDeniedErr(err error) bool {
 	}
 	return s.Code() == codes.PermissionDenied
 }
+
+type CortexMetadataFilter interface {
+	block.DeduplicateFilter
+	block.MetadataFilter
+}
+
+type DisabledDeduplicateFilter struct {
+}
+
+func (f *DisabledDeduplicateFilter) Filter(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, synced block.GaugeVec, modified block.GaugeVec) error {
+	// don't do any deduplicate filtering
+	return nil
+}
+
+func (f *DisabledDeduplicateFilter) DuplicateIDs() []ulid.ULID {
+	return nil
+}
```

Comment:
Can we make these types private? 
Can you add some comment to `DisabledDeduplicateFilter`. We want to disable duplicate filter because it makes no sense for partitioning compactor as we always have duplicates?

Reply:
The `DefaultDeduplicateFilter` from Thanos would mark duplicate blocks if those blocks are from same group and having same source blocks. In partitioning compactor, partitions from same time range would always or eventually have same source as it is the natural of partitioning compactor. We don't want those blocks got filtered out when doing grouping for next level compaction.